taskcluster/ac_taskgraph/loader/build_config.py

```diff
@@ -43,28 +44,40 @@ def get_components_changed(files_changed):
 
 cached_deps = {}
 
-# Gradle is expensive to run. Cache the results for each component.
-@memoize
-def get_deps_for_component(component):
+def get_upstream_deps_for_components(components):
     """Return the full list of local upstream dependencies of a component."""
     deps = set()
+    cmd = ["./gradlew"]
+    for c in sorted(components):
+        cmd.extend(["%s:dependencies" % c, "--configuration", "implementation"])
     # Parsing output like this is not ideal but bhearsum couldn't find a way
     # to get the dependencies printed in a better format. If we could convince
     # gradle to spit out JSON that would be much better.
     # This is filed as https://github.com/mozilla-mobile/android-components/issues/7814
-    for line in subprocess.check_output(['./gradlew', '%s:dependencies' % component, '--configuration', 'implementation']).splitlines():
+    current_component = None
+    for line in subprocess.check_output(cmd).splitlines():
+        # If we find the start of a new component section, update our tracking variable
+        if line.startswith("Project"):
```

Comment:
The parser you wrote is much simpler than the one I had in mind. Excellent! 👍

Reply:
I'm just glad gradle uses some basic ascii formatting to help denote sections :laughing: